src/validation.cpp

```diff
@@ -1005,10 +1005,50 @@ bool GetTransaction(const uint256& hash, CTransactionRef& txOut, const Consensus
     return false;
 }
 
+/**
+ * If the transaction hash is found inside a block, the block hash is placed in hashBlock.
+ * If blockIndex is provided, the transaction is fetched from the corresponding block and its hash is placed in hashBlock.
+ */
+bool GetBlockHash(const uint256& hash, const Consensus::Params& consensusParams, uint256& hashBlock, const CBlockIndex* const blockIndex)
```

Comment:
I feel like this function name should change to something like `GetBlockHashFromTxHash` to distinguish from the existing `GetBlockHash` class member function. And the first input parameter name can change to `hashTx` to easy understand.